src/sentry/grouping/ingest/hashing.py

```diff
@@ -141,16 +147,19 @@ def _calculate_secondary_hashes(
             event_copy = copy.deepcopy(job["event"])
             secondary_hashes = _calculate_event_grouping(
                 project, event_copy, secondary_grouping_config
-            )
+            )[0]
     except Exception as err:
         sentry_sdk.capture_exception(err)
 
-    return secondary_hashes
+    # Return an empty variants dictionary because we need the signature of this function to match
```

Comment:
why does the signature have to match?

Reply:
Okay, so that's half a lie, and half true. The two pairs of functions in question are `_calculate_primary_hashes`/`_calculate_secondary_hashes` and `run_primary_grouping`/`maybe_run_secondary_grouping`. Technically, the signatures of the first pair don't have to match, because they're each only used in the corresponding member of the second pair. The signatures of the secondary pair really do have to match, though, because they're both passed to `get_hashes_and_grouphashes` as `hash_calculation_function`.

https://github.com/getsentry/sentry/blob/779d6a288ecc2fdfd20fc98f4092dd915568c607/src/sentry/event_manager.py#L1357-L1364

It sort of seemed easier to just keep the structures parallel, but you're right, I could be more accurate in how I describe what's happening.

Reply:
yep that makes sense. only Q then is it a bit more explicit to use pythons unwrapping syntax instead of grabbing the 0th element? so it'd be 

```
secondary_hashes, _ = _calculate_event_grouping(
                project, event_copy, secondary_grouping_config
            )
```

Reply:
IDK, do you think it is? It kinda seems six-of-one-half-a-dozen-of-the-other to me. Happy to change it if you think the other is better, since I'm already going back to clean up the comment.

Reply:
i like it because its more explicit, as i was reviewing the PR i went to go make sure "is that the hashes being returned, ah yep okay". but totally a small nit and feel free to keep it the way that it is if you like it better.

Reply:
Ah, I see what you're saying. Okay, sure, I'll switch it.

Reply:
UPDATE: In the end, I decided the most explicit thing would be to leave `_calculate_secondary_hashes` as is (not have it return variants, and not say it's doing so in the name, the way `_calculate_primary_hashes` - now called `_calculate_primary_hashes_and_variants` - does). I did still make `maybe_run_secondary_grouping` return an empty dictionary for the variants (because the matching-signature constraint there is real), but now the inner helpers actually tell the truth about what they do/don't do.

(As you suggested, I did still switch the `[0]` to be unpacking in `_calculate_secondary_hashes`, though.)